engine/gui/controller.py

```diff
@@ -58,3 +67,8 @@ def _activate_current_button(self):
             return
 
         self.selected_button.action(self.api)
+
+    def _change_current_button(self, button_name) -> None:
+        for next_button in self.gui.spec.buttons:
+            if button_name == next_button.selected_image_asset:
```

Comment:
The image asset name is not the same as the button name. You want to compare against the button name, not the asset.